src/views/webviewController.ts

```diff
@@ -49,17 +59,24 @@ export const getWebviewContent = ({
 }): string => {
   const jsAppFileUrl = getReactAppUri(extensionPath, webview);
 
+  // Use a nonce to only allow specific scripts to be run
+  const nonce = getNonce();
+
   return `<!DOCTYPE html>
   <html lang="en">
     <head>
         <meta charset="UTF-8">
+        <meta http-equiv="Content-Security-Policy" content="default-src 'none';
+            script-src vscode-resource: 'self' 'unsafe-inline' 'unsafe-eval' https:; script-src;
+            style-src vscode-resource: 'self' 'unsafe-inline';
+            img-src vscode-resource: 'self' "/>
         <meta name="viewport" content="width=device-width, initial-scale=1.0">
         <title>MongoDB</title>
     </head>
     <body>
       <div id="root"></div>
-      <script>window['${VSCODE_EXTENSION_SEGMENT_ANONYMOUS_ID}'] = '${telemetryUserId}';</script>
-      <script src="${jsAppFileUrl}"></script>
+      <script nonce="${nonce}">window['${VSCODE_EXTENSION_SEGMENT_ANONYMOUS_ID}'] = '${telemetryUserId}';</script>
+      <script nonce="${nonce}" src="${jsAppFileUrl}"></script>
```

Comment:
Not sure if this is a vscode requirement or not and maybe vscode just handles it different to real browser, but usually setting a `nonce` [requires to provide the same nonce as part of the `Content-Security-Policy`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/nonce) and based on the CSP above I don't think this nonce will do anything?